api/optimeet/groups/views.py

```diff
@@ -63,6 +63,28 @@ def add_users_to_group(request, group_id):
     
     return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+@csrf_exempt
+@api_view(['POST'])
+def add_preference_to_group(request, group_id, user_id, format=None):
+        # Check if the user group exists
+    try:
+        group = models.Group.objects.get(group_id=group_id)
+        
+        user_group = models.UserGroup.objects.get(group_id=group_id, user_id=user_id)
+
+    except models.UserGroup.DoesNotExist:
```

Comment:
Let's also add an exception for the Group not existing

Reply:
done.